plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/EqualityDeleteSets.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.util.StructLikeSet;
+import org.apache.iceberg.util.StructProjection;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class EqualityDeleteSets
+{
+    private final Map<Set<Integer>, EqualityDeleteSet> deleteSetById = new HashMap<>();
+
+    public StructLikeSet getDeleteSet(Set<Integer> ids, Schema schema)
+    {
+        EqualityDeleteSet equalityDeleteSet = deleteSetById.computeIfAbsent(ids, key -> new EqualityDeleteSet(schema, StructLikeSet.create(schema.asStruct())));
+        return equalityDeleteSet.getDeleteSet();
+    }
+
+    public void initialize(Schema fileSchema)
+    {
+        deleteSetById.values().forEach(equalityDeleteSet -> equalityDeleteSet.initialize(fileSchema));
+    }
+
+    public boolean isEmpty()
+    {
+        return deleteSetById.isEmpty();
+    }
+
+    public boolean contains(StructLike row)
+    {
+        return deleteSetById.values().stream().anyMatch(equalityDeleteSet -> equalityDeleteSet.contains(row));
+    }
```

Comment:
Most of this code can be actually inlined in the method `io.trino.plugin.iceberg.IcebergPageSourceProvider#readDeletes` without adding extra complexity to the readability of the method.